cmd/kyma/alpha/create/scaffold/scaffold.go

```diff
@@ -0,0 +1,286 @@
+	)
+
+	cmd.Flags().BoolVar(
+		&o.GenerateSecurityConfig, "gen-sec-config", false,
+		"Specifies if security config should be generated",
+	)
+	cmd.Flags().BoolVar(
+		&o.GenerateManifest, "gen-manifest", false,
+		"Specifies if manifest file should be generated",
+	)
+	cmd.Flags().BoolVar(
+		&o.GenerateDefaultCR, "gen-default-cr", false,
+		"Specifies if a default CR should be generated",
+	)
+
+	cmd.Flags().StringVar(
+		&o.ModuleConfigName, "module-name", "",
+		"Specifies the module name in the generated module config file",
+	)
+	cmd.Flags().StringVar(
+		&o.ModuleConfigVersion, "module-version", "",
+		"Specifies the module version in the generated module config file",
+	)
+	cmd.Flags().StringVar(
+		&o.ModuleConfigChannel, "module-channel", "",
+		"Specifies the module channel in the generated module config file",
+	)
+	cmd.Flags().StringVar(
+		&o.ModuleConfigManifestPath, "module-manifest-path", "",
+		"Specifies the module manifest filepath in the generated module config file",
+	)
+
+	return cmd
+}
+
+func (cmd *command) Run(_ context.Context) error {
+	if cmd.opts.CI {
+		cmd.Factory.NonInteractive = true
+	}
+	if cmd.opts.Verbose {
+		cmd.Factory.UseLogger = true
+	}
+
+	l := cli.NewLogger(cmd.opts.Verbose).Sugar()
+	undo := zap.RedirectStdLog(l.Desugar())
+	defer undo()
+
+	if !cmd.opts.NonInteractive {
+		cli.AlphaWarn()
+	}
+
+	if err := cmd.opts.Validate(); err != nil {
+		return err
+	}
+
+	if cmd.opts.GenerateManifest {
+		cmd.NewStep("Generating manifest file...")
+
+		err := cmd.generateManifest()
+		if err != nil {
+			cmd.CurrentStep.Failuref("%v: %w", errManifestCreationFailed, err)
+			return fmt.Errorf("%v: %w", errManifestCreationFailed, err)
+		}
+
+		cmd.CurrentStep.Successf("Generated %s", fileNameManifest)
+	}
+
+	if cmd.opts.GenerateSecurityConfig {
+		cmd.NewStep("Generating security config file...")
+
+		err := cmd.generateSecurityConfig()
+		if err != nil {
+			cmd.CurrentStep.Failuref("%w: %v", errSecurityConfigCreationFailed, err)
+			return fmt.Errorf("%w: %v", errSecurityConfigCreationFailed, err)
+		}
+
+		cmd.CurrentStep.Successf("Generated %s", fileNameSecurityConfig)
+	}
+
+	if cmd.opts.GenerateDefaultCR {
+		cmd.NewStep("Generating default CR file...")
+
+		err := cmd.generateDefaultCR()
+		if err != nil {
+			cmd.CurrentStep.Failuref("%w: %v", errDefaultCRCreationFailed, err)
+			return fmt.Errorf("%w: %v", errDefaultCRCreationFailed, err)
+		}
+
+		cmd.CurrentStep.Successf("Generated %s", fileNameDefaultCR)
+	}
+
+	cmd.NewStep("Generating module config file...")
+
+	err := cmd.generateModuleConfig()
+	if err != nil {
+		cmd.CurrentStep.Failuref("%w: %v", errModuleConfigCreationFailed, err)
+		return fmt.Errorf("%w: %v", errModuleConfigCreationFailed, err)
+	}
+
+	cmd.CurrentStep.Successf("Generated %s", fileNameModuleConfig)
+	return nil
+}
+
+func (cmd *command) generateManifest() error {
+	makeCmd := exec.Command("make", "build-manifests")
```

Comment:
This is to be discussed. It requires a Makefile with a specific target.
First: this contract is not described in the ACs
Second: it's not documented